frontend/src/app/Settings/Facility/FacilityForm.tsx

```diff
@@ -286,37 +244,78 @@ const FacilityForm: React.FC<Props> = (props) => {
     }
   };
 
+  const getUpdatedValues = (
+    addresses?: Partial<Record<AddressOptions, AddressWithMetaData>>
+  ) => {
+    const values = getValues();
+    let adjustedFacility: FacilityFormData;
+
+    if (addresses) {
+      adjustedFacility = {
+        facility: {
+          ...values.facility,
+          ...addresses.facility,
+        },
+        orderingProvider: {
+          ...values.orderingProvider,
+          ...addresses.provider,
+        },
+        devices: values.devices || [],
+      };
+    } else {
+      adjustedFacility = {
+        facility: {
+          ...values.facility,
+        },
+        orderingProvider: {
+          ...values.orderingProvider,
+        },
+        devices: values.devices || [],
+      };
+    }
+
+    return adjustedFacility;
+  };
+
   const updateAddressesAndSave = (
     addresses: Partial<Record<AddressOptions, AddressWithMetaData>>
   ) => {
-    const adjustedFacility: Facility = {
-      ...facility,
-      ...addresses.facility,
-      orderingProvider: {
-        ...facility.orderingProvider,
-        ...addresses.provider,
-      },
-    };
-    updateFormData(adjustedFacility);
+    const adjustedFacility = getUpdatedValues(addresses);
+    setValue("facility", adjustedFacility.facility);
+    setValue("orderingProvider", adjustedFacility.orderingProvider);
     props.saveFacility(adjustedFacility);
   };
 
-  const validateAndSaveFacility = async () => {
-    if ((await validateFacility()) === "error") {
-      return;
-    }
+  const onSubmit = async (facilityData: FacilityFormData) => {
+    const { facility, orderingProvider, devices } = facilityData;
+    setValue("facility", facility);
+    setValue("orderingProvider", orderingProvider);
+    setValue("devices", devices);
+    clearErrors();
     await validateFacilityAddresses();
   };
 
+  const onError = () => {
+    showError(
+      "Please check the form to make sure you complete all of the required fields.",
```

Comment:
similar note to the comment [here](https://github.com/CDCgov/prime-simplereport/pull/5983/files#r1235287974)